Fix clearing a selection in a grid with reordered columns

thesiv opened the pull request:
`wxGrid::GetColPos` must not be called with invalid index (-1) so fix
`wxGrid::RefreshBlock` to pass to `GetColPos` only valid indices.

thesiv committed:
Fix clearing a selection in a grid with reordered columns

`wxGrid::GetColPos` must not be called with invalid index (-1) so fix
`wxGrid::RefreshBlock` to pass to `GetColPos` only valid indices.

Add a unit test verifying that this works as intended.

vadz reviewed (commented):
I think the change is correct, strictly speaking, but I still have some problems with this function (`RefreshBlock()`):

1. I don't understand what its preconditions are: clearly, we may have `topLeft = (-1, -1)` or `bottomRight = (-1, -1)` on input. But can we have `topRow = -1` and `leftCol != -1` (and the same thing for bottom right)? IMO this shouldn't be possible, but the assert code seems to assume that it is, as it checks for `topRow != -1 || leftCol != -1` where it could be checking for `topRow != -1 && leftCol != -1` instead. It would be great if we could document this clearly.
1. There are no tests, which makes it difficult to know how is the function supposed to behave and be sure that no bugs are introduced by changes to it. Could you please add some tests for it to `tests/controls/gridtest.cpp`?

TIA!

thesiv committed:
fixup! Fix clearing a selection in a grid with reordered columns

thesiv commented:
Fixed, added the unit test and tested undex Win, Ubuntu (GTK3) and Mac (but `test_gui` is crashed when run with WX_UI_TESTS=1 with or without the new added test).
